ovh/resource_cloud_project_instance.go

```diff
@@ -0,0 +1,387 @@
+package ovh
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/url"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/ovh/go-ovh/ovh"
+)
+
+func resourceCloudProjectInstance() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceCloudProjectInstanceCreate,
+		ReadContext:   resourceCloudProjectInstanceRead,
+		DeleteContext: resourceCloudProjectInstanceDelete,
+
+		Schema: map[string]*schema.Schema{
+			"service_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				DefaultFunc: schema.EnvDefaultFunc("OVH_CLOUD_PROJECT_SERVICE", nil),
+				Description: "Service name of the resource representing the id of the cloud project.",
+				ForceNew:    true,
+			},
+			"region": {
+				Type:        schema.TypeString,
+				Description: "Instance region",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"auto_backup": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "Create an autobackup workflow after instance start up",
+				ForceNew:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"cron": {
+							Type:        schema.TypeString,
+							Description: "Unix cron pattern",
+							Optional:    true,
+							ForceNew:    true,
+						},
+						"rotation": {
+							Type:        schema.TypeInt,
+							Description: "Number of backup to keep",
+							Optional:    true,
+							ForceNew:    true,
+						},
+					},
+				},
+			},
+			"billing_period": {
+				Type:        schema.TypeString,
+				Description: "Number of backup to keep",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"boot_from": {
+				Type:        schema.TypeList,
+				Required:    true,
+				Description: "Boot the instance from an image or a volume",
+				ForceNew:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"image_id": {
+							Type:        schema.TypeString,
+							Description: "Instance image id",
+							Optional:    true,
+						},
+						"volume_id": {
+							Type:        schema.TypeString,
+							Description: "Instance volume id",
+							Optional:    true,
+						},
+					},
+				},
+			},
+			"bulk": {
+				Type:        schema.TypeInt,
+				Description: "Create multiple instances",
+				Optional:    true,
+				ForceNew:    true,
+			},
+			"flavor": {
+				Type:        schema.TypeList,
+				Required:    true,
+				Description: "Flavor information",
+				ForceNew:    true,
```

Comment:
is this really true ? It seems that right now with openstack provider you can change flavor without recreating

Reply:
will be in v2